src/pedal/config.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pedals: Vec<Pedal>,
+}
+
+impl Config {
+    pub fn from(yaml: &str) -> Result<Config> {
+        Ok(serde_yaml::from_str(yaml)?)
+    }
+
+    pub fn default() -> Self {
+        Self {
+            pedals: vec![Transparent],
+        }
+    }
+
+    pub fn to_pedals(&self, stream_config: &StreamConfig) -> Result<Vec<pedal::Boxed>> {
+        self.pedals
+            .iter()
+            .map(|pedal| pedal.to_pedal(stream_config))
+            .collect()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
```

Comment:
This is telling `serde` that it should expect the enums to be serialized like this:

```yaml
- type: EnumCaseName
  other_case_specific_field_1: 1
  other_case_specific_field_2: "abc"
```

So all `Pedal` cases have a `type` "tag" field in common, and then the rest of the fields are determined by the type of each case.